rfc/tap-hardening.md

```diff
@@ -0,0 +1,99 @@
+# Summary
+
+The goal of the Tap Hardening project is to restrict the set of clients that can
```

Comment:
Maybe pedantic, but `restrict the set of users by integrating with native Kubernetes RBAC`.

Reply:
I agree a good thing to be explicit about when the design is focused on using Kubernetes primitives!